File: DSA_Codesheet/Python/insertion_Sort.py
# The following function performs insertion sort in ascending order.
# The datatype used is a list(array of numbers).


def insertion_sort(arr):
  for i in range(1, len(arr)):
    key = arr[i]
    j = i - 1
    while j >= 0 and key < arr[j]:
      arr[j + 1] = arr[j]
      j -= 1
    arr[j + 1] = key

  return arr



arr = input("Enter a list of numbers: ").split(" ")

arr = [int(i) for i in arr]

sorted_arr = insertion_sort(arr)

for i in sorted_arr:
  print(i,end=' ')

File: DSA_Codesheet/Java/InsertionSort.java
public class InsertionSort {

    public static void insertionSort(int arr[]) {
        int n = arr.length;
        for (int i = 1; i < n; ++i) {
            int key = arr[i];
            int j = i - 1;

            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }


    public static void main(String args[]) {
        int arr[] = {18, 0, 11, 1, 24, 91, 100};
        int n = arr.length;

        insertionSort(arr);

        for (int i = 0; i < n; ++i)
        System.out.print(arr[i] + " ");
        
        System.out.println();
        
    }
}
